src/docfx/build/moniker/MonikersProvider.cs

```diff
@@ -8,14 +8,14 @@
 
 namespace Microsoft.Docs.Build
 {
-    internal class MonikersProvider
+    internal class MonikerProvider
     {
         private readonly List<(Func<string, bool> glob, string monikerRange)> _rules = new List<(Func<string, bool>, string)>();
         private readonly MonikerRangeParser _rangeParser;
 
         public MonikerComparer Comparer { get; }
 
-        public MonikersProvider(Docset docset)
+        public MonikerProvider(Docset docset)
```

Comment:
i am a little confused with MonikerProvider and MonikerMap, seems they are all used to provide monikers, why not combine them together, as i know, all monikers are provided by moniker provider, so moniker provider should already know the mapping

Reply:
as we talked offline, it would be great if you can only have one `monikerResolver` to
1. provider moniker
2. tracing moniker dependency

then we don't need add moniker map everywhere and also don't need return monikers everywhere, just like DependencyResolver